src/components/course/repo/lessons.repo.integration.test.js

```diff
@@ -0,0 +1,63 @@
+const chai = require('chai');
+const chaiAsPromised = require('chai-as-promised');
+
+const { withApp, testObjects } = require('../../../../test/utils/withApp.test');
+const lessonsRepo = require('./lessons.repo');
+
+const { equal } = require('../../../helper/compare').ObjectId;
+
+chai.use(chaiAsPromised);
+const { expect } = chai;
+
+const getTestObjects = async () => {
+	const courseTeacher = await testObjects.createTestUser({ roles: 'teacher' });
+	const course = await testObjects.createTestCourse({ teacherIds: [courseTeacher._id] });
+	const contents = [
+		{
+			user: courseTeacher._id,
+			component: 'text',
+			title: 'contents title',
+			content: { text: 'some text content' },
+			hidden: false,
+		},
+	];
+	const lesson = await testObjects.createTestLesson({
+		name: 'Demo lesson with user',
+		courseId: course._id,
+		contents,
+	});
+	return { courseTeacher, lesson };
+};
+
+describe(
+	'when having a user in lesson contents',
+	withApp(() => {
+		it('should remove the user related lesson contents', async () => {
+			// create test objects
+			const { courseTeacher } = await getTestObjects();
+
+			// delete user from lesson contents
+			const result = await lessonsRepo.deleteUserFromLessonContents(courseTeacher._id);
+			expect(result.matchedDocuments, 'only one lesson should match').to.equal(1);
+			expect(result.modifiedDocuments, 'only one lesson should be modified').to.equal(1);
+
+			// the user has been removed
+			const matchesAfter = await lessonsRepo.getLessonsWithUserInContens(courseTeacher._id);
+			expect(matchesAfter.length, 'all results have been removed').to.equal(0);
```

Comment:
i like the notation should  in it and the when in expect  'when all results .. but is only personal preferences.